server/src/test/java/org/elasticsearch/cluster/metadata/MetaStateTests.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.common.UUIDs;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class MetaStateTests extends ESTestCase {
+
+    private MetaState copyState(MetaState state, boolean introduceErrors){
+        long generation = state.getGlobalStateGeneration();
+        Map<Index, Long> indices = new HashMap<>(state.getIndices());
+        if (introduceErrors) {
+            if (randomBoolean()){
+                generation = generation + 1;
+            } else {
+                indices.remove(randomFrom(indices.keySet()));
```

Comment:
We should also have tests that add an index and ones that change an index's generation.

Reply:
++

Reply:
https://github.com/elastic/elasticsearch/pull/35199/commits/1eee534f081c804c7081aa66ac03fafbc301d4e3